kube-rbac-proxy.yaml

```diff
@@ -65,18 +69,94 @@ test:
+          apiGroup: rbac.authorization.k8s.io
+          kind: ClusterRole
+          name: kube-rbac-proxy
+        subjects:
+        - kind: ServiceAccount
+          name: default
+          namespace: default
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: ClusterRole
+        metadata:
+          name: kube-rbac-proxy
+        rules:
+        - apiGroups: ["authentication.k8s.io"]
+          resources:
+          - tokenreviews
+          verbs: ["create"]
+        - apiGroups: ["authorization.k8s.io"]
+          resources:
+          - subjectaccessreviews
+          verbs: ["create"]
+        YAML
+    - name: "Setup kube-rbac-proxy"
+      runs: |
+        # Start an echo server
+        http-echo -listen=:9999 -text="hello world" > /dev/null 2>&1 &
+        # Start kubeproxy
+        kube-rbac-proxy \
+          --upstream=http://127.0.0.1:9999 \
+          --secure-listen-address=127.0.0.1:8081 \
+          --tls-cert-file /etc/tls-key-cert-pair/tls.crt \
+          --tls-private-key-file /etc/tls-key-cert-pair/tls.key > /dev/null 2>&1 &
+        wait-for-it 127.0.0.1:8081 --timeout=60
+    - name: "Checking kube-rbac-proxy without rbac"
+      runs: |
+        response=$(curl --connect-timeout 5 -s -o /dev/null -k -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" -w "%{http_code}" https://localhost:8081/)
+        if [ "$response" -ne 403 ]; then
+          echo "kube-rbac-proxy returned wrong status code: want=403 got=$response"
+          exit 1
+        fi
+    - name: "Creating RBAC"
+      runs: |
+        kubectl apply -f - <<YAML
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: ClusterRole
+        metadata:
+          name: access
+        rules:
+        - nonResourceURLs: ["/"]
+          verbs: ["get"]
+        ---
+        apiVersion: rbac.authorization.k8s.io/v1
+        kind: ClusterRoleBinding
+        metadata:
+          name: access
+        roleRef:
+          apiGroup: rbac.authorization.k8s.io
+          kind: ClusterRole
+          name: access
+        subjects:
+        - kind: ServiceAccount
+          name: default
+          namespace: default
+        YAML
+    - name: "Force restart kube-rbac-proxy"
+      runs: |
+        # It can take some time for the binary to pick the new RBAC. This is fast and predicable.
+        kill -9 $(pgrep kube-rbac-proxy)
+        kube-rbac-proxy \
+          --upstream=http://127.0.0.1:9999 \
+          --secure-listen-address=127.0.0.1:8081 \
+          --tls-cert-file /etc/tls-key-cert-pair/tls.crt \
+          --tls-private-key-file /etc/tls-key-cert-pair/tls.key > /dev/null 2>&1 &
+        wait-for-it 127.0.0.1:8081 --timeout=60
+    - name: "Checking kube-rbac-proxy with RBAC"
+      runs: |
+        response=$(curl --connect-timeout 5 -s -o /dev/null -k -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" -w "%{http_code}" https://localhost:8081/)
```

Comment:
any reason to use `-k` here, I think we should omit that given we're hitting a https endpoint

Reply:
@kranurag7 I was just being lazy. updated